src/measure_theory/function/continuous_map_dense.lean

```diff
@@ -42,152 +58,332 @@ Vitali-Carathéodory theorem, in the file `measure_theory.vitali_caratheodory`.
+  have I1 : snorm (s.indicator (λ y, c) - t.indicator (λ y, c)) p μ ≤ ε/2,
+  { rw [← snorm_neg, neg_sub, ← indicator_diff st],
+    exact (hη _ μs.le) },
+  rcases exists_compact_between s_compact is_open_univ (subset_univ _) with ⟨k, k_compact, sk, -⟩,
+  rcases exists_continuous_snorm_sub_le_of_closed hp s_compact.is_closed is_open_interior sk
+    hsμ.ne c h'ε with ⟨f, I2, f_cont, f_bound, f_support, f_mem⟩,
+  have I3 : snorm (f - t.indicator (λ y, c)) p μ ≤ ε, from calc
+    snorm (f - t.indicator (λ y, c)) p μ
+      = snorm ((f - s.indicator (λ y, c)) + (s.indicator (λ y, c) - t.indicator (λ y, c))) p μ :
+    by simp only [sub_add_sub_cancel]
+  ... ≤ snorm (f - s.indicator (λ y, c)) p μ
+        + snorm (s.indicator (λ y, c) - t.indicator (λ y, c)) p μ :
+    begin
+      apply snorm_add_le _ _ h'p,
+      { exact f_mem.ae_strongly_measurable.sub
+          (ae_strongly_measurable_const.indicator s_compact.measurable_set) },
+      { exact (ae_strongly_measurable_const.indicator s_compact.measurable_set).sub
+          (ae_strongly_measurable_const.indicator ht) },
+    end
+  ... ≤ ε/2 + ε/2 : add_le_add I2 I1
+  ... = ε : ennreal.add_halves _,
+  refine ⟨f, I3, f_cont, f_mem, _⟩,
+  apply has_compact_support.intro k_compact (λ x hx, _),
+  rw ← function.nmem_support,
+  contrapose! hx,
+  exact interior_subset (f_support hx)
+end
+
+/-- In a locally compact space, any function in `ℒp` can be approximated by compactly supported
+continuous functions when `1 ≤ p < ∞`, version in terms of `∫`. -/
+lemma mem_ℒp.exists_has_compact_support_integral_rpow_sub_le
+  [locally_compact_space α] [μ.regular] {p : ℝ} (h'p : 1 ≤ p)
+  {f : α → E} (hf : mem_ℒp f (ennreal.of_real p) μ) {ε : ℝ} (hε : 0 < ε) :
+  ∃ (g : α → E), ∫ x, ‖f x - g x‖^p ∂μ ≤ ε ∧ continuous g ∧ mem_ℒp g (ennreal.of_real p) μ
+    ∧ has_compact_support g :=
+begin
+  have I : 0 < ε ^ (1/p) := real.rpow_pos_of_pos hε _,
+  have A : ennreal.of_real (ε ^ (1/p)) ≠ 0,
+    by simp only [ne.def, ennreal.of_real_eq_zero, not_le, I],
+  have B : 1 ≤ ennreal.of_real p,
+  { convert ennreal.of_real_le_of_real h'p, exact ennreal.of_real_one.symm },
+  rcases hf.exists_has_compact_support_snorm_sub_le ennreal.coe_ne_top B A
+    with ⟨g, hg, g_cont, g_mem, g_support⟩,
+  change snorm _ (ennreal.of_real p) _ ≤ _ at hg,
+  refine ⟨g, _, g_cont, g_mem, g_support⟩,
+  rwa [(hf.sub g_mem).snorm_eq_integral_rpow_norm (zero_lt_one.trans_le B).ne'
+    ennreal.coe_ne_top, ennreal.of_real_le_of_real_iff I.le, one_div,
+    ennreal.to_real_of_real (zero_le_one.trans h'p), real.rpow_le_rpow_iff _ hε.le _] at hg,
+  { exact integral_nonneg (λ x, real.rpow_nonneg_of_nonneg (norm_nonneg _) _) },
+  { exact inv_pos.2 (zero_lt_one.trans_le h'p) }
+end
+
+/-- In a locally compact space, any integrable function can be approximated by compactly supported
+continuous functions, version in terms of `∫⁻`. -/
+lemma integrable.exists_has_compact_support_lintegral_sub_le [locally_compact_space α] [μ.regular]
```

Comment:
Maybe not, but who knows. Often when dealing with norms, the ennreal-valued integral is nicer to use than the real one. Since it's really short, I'd rather keep it.